pkg/codegen/nodejs/gen.go

```diff
@@ -120,6 +120,25 @@ func (mod *modContext) tokenToType(tok string, input bool) string {
 	return root + modName + title(name)
 }
 
+func (mod *modContext) tokenToResource(tok string) string {
+	// token := pkg : module : member
+	// module := path/to/module
+
+	components := strings.Split(tok, ":")
+	contract.Assertf(len(components) == 3, "malformed token %v", tok)
+
+	modName, name := mod.pkg.TokenToModule(tok), title(components[2])
+	if override, ok := mod.modToPkg[modName]; ok {
+		modName = override
+	}
```

Comment:
Nit: it'd be nice to factor this piece out into a function as it's shared with `tokenToType`.